### PR TITLE
fix allow to search selfDM in new chat page

### PR DESCRIPTION
**Pull request by nkdengineer**
### Fixed Issues

$ https://github.com/Expensify/App/issues/40259
PROPOSAL: https://github.com/Expensify/App/issues/40259#issuecomment-2057972070

### Tests

1. Press FAB then Start chat
2. In the search field, type your email, your name or the word "you" ("tù" if language is set to Spanish)
3. Verify you see the self dm report (a report with your name and avatar)
4. Press the self dm report
5. Verify you are navigated to the self dm report
- [x] Verify that no errors appear in the JS console

### QA Steps

1. Press FAB then Start chat
2. In the search field, type your email, your name or the word "you" ("tù" if language is set to Spanish)
3. Verify you see the self dm report (a report with your name and avatar)
4. Press the self dm report
5. Verify you are navigated to the self dm report
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/161821005/39c39459-9833-43a9-b4d5-5b16253db775

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/161821005/2670e84c-ab4f-4a0b-b408-a29e9070e31a

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/161821005/217f12d4-ddc4-4b61-bd8d-a9226c10ec3d

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/161821005/01cfae41-b174-4651-9e21-26abba7c4c18

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/161821005/b11c92b4-fd12-41cf-af02-1b9d2e4f711e

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/161821005/a3f398d6-3606-455b-b2ae-d6502a21115a

</details>

### CODE REVIEW

File: src/pages/NewChatPage.tsx [hunk 1]
```diff
@@ -182,6 +186,11 @@ function NewChatPage({isGroupChat}: NewChatPageProps) {
      */
     const createChat = useCallback(
         (option?: OptionsListUtils.Option) => {
+            const report = getReport(option?.reportID);
+            if (option?.isSelfDM && report) {
+                Navigation.dismissModalWithReport(report);
```
Comment (s77rt) [inline src/pages/NewChatPage.tsx:191]: Use dismissModal. Internally it will get the report

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by nkdengineer** [comment]
I am updating the videos.

File: src/pages/NewChatPage.tsx [hunk 2]
```diff
@@ -182,6 +187,11 @@ function NewChatPage({isGroupChat}: NewChatPageProps) {
      */
     const createChat = useCallback(
         (option?: OptionsListUtils.Option) => {
+            const report = ReportUtils.getReport(option?.reportID);
+            if (option?.isSelfDM && report) {
+                Navigation.dismissModal(report.reportID);
```
Comment (s77rt) [inline src/pages/NewChatPage.tsx:192]: Use option.reportID

File: src/libs/OptionsListUtils.ts [hunk 3]
```diff
@@ -1741,7 +1742,10 @@ function getOptions(
     }
 
     // Exclude the current user from the personal details list
-    const optionsToExclude: Option[] = [{login: currentUserLogin}, {login: CONST.EMAIL.NOTIFICATIONS}];
+    const optionsToExclude: Option[] = [{login: CONST.EMAIL.NOTIFICATIONS}];
+    if (!includeSelfDM) {
+        optionsToExclude.push({login: currentUserLogin});
+    }
```
Comment (nkdengineer) [inline src/libs/OptionsListUtils.ts:1748]: @s77rt Using like this leads to the jest test 2 is failed

Comment (nkdengineer) [reply src/libs/OptionsListUtils.ts:1748]: If I revert that change and update [this](https://github.com/Expensify/App/blob/76ca7814304e143c86d4a2031c69af8aba8e17a9/src/libs/OptionsListUtils.ts#L1774) to:
```
 if (!reportOption.isSelfDM &&
                (!includeThreads &&
                    (!!reportOption.login || reportOption.reportID) &&
                    optionsToExclude.some((option) => option.login === reportOption.login || option.reportID === reportOption.reportID))
            ) {
```
the test is passed

Comment (s77rt) [reply src/libs/OptionsListUtils.ts:1748]: I'm not seeing any failing test

**Comment by s77rt** [comment]
Please change the testing steps to:

1. Press FAB then Start chat
2. In the search field, type your email, your name or the word "you" ("tù" if language is set to Spanish)
3. Verify you see the self dm report (a report with your name and avatar)
4. Press the self dm report
5. Verify you are navigated to the self dm report

**Comment by s77rt** [comment]
@nkdengineer Can you please add the missing screenshots

**Comment by s77rt** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


android

</details>

<details>
<summary>Android: mWeb Chrome</summary>


mweb-chrome

</details>

<details>
<summary>iOS: Native</summary>


ios

</details>

<details>
<summary>iOS: mWeb Safari</summary>


mweb-safari

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


web

</details>

<details>
<summary>MacOS: Desktop</summary>


desktop

</details>

**Comment by s77rt** [comment]
Please tag me once the above concerns are addressed

File: src/libs/OptionsListUtils.ts (hunk 3 above)
Comment (nkdengineer) [reply src/libs/OptionsListUtils.ts:1748]: Here is the test result https://github.com/Expensify/App/actions/runs/8780426190/job/24090318554?pr=40672. can you check it?

**Comment by nkdengineer** [comment]
@s77rt All the videos and test steps are updated. Just need to address the issue https://github.com/Expensify/App/pull/40672#discussion_r1574417096

**Comment by s77rt** [comment]
I checked this and it seems the comments and the code do not match (outdated comments). We want to exclude the current user login from the personal details list only and not the reports list. The reports list already works es expected because If includeSelfDM is false, the user won't be even included in the reports list => we don't need to exclude him / nothing to exclude.

In the reports list however we want to exclude selected options, even if that option is a self dm. [Your solution](https://github.com/Expensify/App/pull/40672#discussion_r1574425408) will break this as it will make the self dm always present even when we want to exclude it (if it's selected)

Suggested solution:
1. Define optionsToExclude as follow: `const optionsToExclude: Option[] = [{login: CONST.EMAIL.NOTIFICATIONS}];`
2. In personal details list loop extend the [`optionsToExlucde`](https://github.com/Expensify/App/blob/944751fcec525ece0da16a6a4efd80ad8ab4ff8c/src/libs/OptionsListUtils.ts#L1825) to cover the current user login: `[...optionsToExclude, {login: currentUserLogin}]`

**Comment by nkdengineer** [comment]
@s77rt Thanks for your suggestion. I updated PR based on that.

File: src/libs/OptionsListUtils.ts [hunk 4]
```diff
@@ -1819,6 +1820,7 @@ function getOptions(
         }
     }
 
+    optionsToExclude.push({login: currentUserLogin});
```
Comment (s77rt) [inline src/libs/OptionsListUtils.ts:1823]: Do not push to optionsToExclude. Instead create a new variable 
```js
const personalDetailsOptionsToExclude = [...optionsToExclude, {login: currentUserLogin}];
```

Define this inside the `if (includePersonalDetails) { ... }` block

**Review by s77rt — CHANGES_REQUESTED** [review]

File: src/libs/OptionsListUtils.ts (hunk 4 above)
Comment (s77rt) [reply src/libs/OptionsListUtils.ts:1823]: Also remove this comment https://github.com/Expensify/App/blob/f60a61093e447477b14b65a5ef96caa3df36bbb6/src/libs/OptionsListUtils.ts#L1744

**Comment by nkdengineer** [comment]
@s77rt I resolved [comment](https://github.com/Expensify/App/pull/40672#discussion_r1576950772)

**Review by s77rt — APPROVED** [review]

**Review by thienlnam — APPROVED** [review]
Great, thanks!

**Pull request merged**